src/components/Message/Message.stories.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { Meta, StoryObj } from '@storybook/react'
+import { useState } from 'react'
+
+import { Hierarchy, Size } from '../Button/Button.types'
+import { Button } from '../Button'
+import { Icon } from '../Icon'
+import { Message } from '.'
+import { useFeedbackMessage } from '../../hooks/useFeedbackMessage/useFeedbackMessage'
+
+const meta = {
+  args: {
+    message: 'This is a Feedback message',
+  },
+  component: Message,
+} satisfies Meta<typeof Message>
+
+export default meta
+type Story = StoryObj<typeof meta>
+
+export const FeedbackMessage: Story = {
+  decorators: [(_Story, context) => {
+    const { info, loading, success, warning, error } = useFeedbackMessage()
+
+    const { message } = context.args
+
+    return (
+      <div style={{ display: 'flex', gap: '4px' }}>
+        <Button
+          hierarchy={Hierarchy.Neutral}
+          icon={<Icon color="blue" name="PiInfo" size={16} />}
+          onClick={() => info({ message })}
+        >
+Info
+        </Button>
+        <Button
+          hierarchy={Hierarchy.Neutral}
+          icon={<Icon color="green" name="PiCheck" size={16} />}
+          onClick={() => success({ message })}
+        >
+Success
+        </Button>
+        <Button
+          hierarchy={Hierarchy.Neutral}
+          icon={<Icon name="PiSpinner" size={16} />}
+          onClick={() => loading({ message })}
+        >
+Loading
+        </Button>
+        <Button
+          hierarchy={Hierarchy.Neutral}
+          icon={<Icon color="orange" name="PiWarning" size={16} />}
+          onClick={() => warning({ message })}
+        >
+Warning
+        </Button>
+        {/* Use icons in btn */}
+        <Button
+          hierarchy={Hierarchy.Neutral}
+          icon={<Icon color="red" name="PiStop" size={16} />}
+          onClick={() => error({ message })}
+        >
+Error
+        </Button>
+      </div >
+    )
+  }],
+}
+
+export const FeedbackMessageWithExtraContent: Story = {
+  decorators: [() => {
+    const { success, dismiss } = useFeedbackMessage()
+
+    const onDismiss = (): void => { dismiss('messageKey') }
+
+    const onClick = (): void => {
+      success({
+        duration: 0,
```

Comment:
There's another issue there: `sticky` should be used instead of `duration: 0` in case the message is not automatically dismissed.

Regarding the highlighted issue, the idea of this message is that user have to dismiss it after the click on the button.

Removed `duration: 0` in favor of `sticky: true`. Change the label of the internal button from "Close" to "Dismiss" to make it easier to understand.

Reply:
However, the loading message remains until the user decides to turn it into a success message (which is not super immediate). At least on that one, I would specify a duration

Reply:
The loading message example has been changed to have a 10 seconds duration, as explained in the other comment.

However, in this example, the message remains sticky to show how messages can be manually dismissable.